src/common/netmap.c

```diff
@@ -549,25 +549,17 @@ bool netmap_tx_queues_empty(void *p)
     assert(sp);
 
     sp->cur_tx_ring = 0;
-    txring = NETMAP_TXRING(sp->nm_if, sp->cur_tx_ring);
-    while (NETMAP_TX_RING_EMPTY(txring)) {
-        /* current ring is empty- go to next */
-        ++sp->cur_tx_ring;
-        if (sp->cur_tx_ring > sp->last_tx_ring) {
-            /* last ring */
-            sp->cur_tx_ring = 0;
-            return true;
-        }
 
-        txring = NETMAP_TXRING(sp->nm_if, sp->cur_tx_ring);
+    for (int i = sp->cur_tx_ring; i <= sp->last_tx_ring; i++) {
```

Comment:
Thanks for the PR. Is this change something that affects all versions of netmap? I am concerned about backwards compatibility, i.e. do we have to #ifdef or check for netmap version numbers?

Reply:
Sorry, as it seems this is not enough to fix the issue... I will investigate further